homework02/allan/string.h

```diff
@@ -0,0 +1,41 @@
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+
+class String
+{
+public:
+    String();
+    String(const char *);
+    String(const String& other);
+                String(const char *, int capactiy);
```

Comment:
Its a convention to first define all ctors, then copy ctor, etc. Copy ctor is a special ctor that should not lie in between the other ones.

Reply:
Will change